exp/addons/internal/controllers/clusterresourceset_controller.go

```diff
@@ -314,54 +311,25 @@ func (r *ClusterResourceSetReconciler) ApplyClusterResourceSet(ctx context.Conte
 			Applied:         false,
 			LastAppliedTime: &metav1.Time{Time: time.Now().UTC()},
 		})
-		// Since maps are not ordered, we need to order them to get the same hash at each reconcile.
-		keys := make([]string, 0)
-		data, ok := unstructuredObj.UnstructuredContent()["data"]
-		if !ok {
-			errList = append(errList, errors.New("failed to get data field from the resource"))
-			continue
-		}
-
-		unstructuredData := data.(map[string]interface{})
-		for key := range unstructuredData {
-			keys = append(keys, key)
-		}
-		sort.Strings(keys)
-
-		dataList := make([][]byte, 0)
-		for _, key := range keys {
-			val, ok, err := unstructured.NestedString(unstructuredData, key)
-			if !ok || err != nil {
-				errList = append(errList, errors.New("failed to get value field from the resource"))
-				continue
-			}
-
-			byteArr := []byte(val)
-			// If the resource is a Secret, data needs to be decoded.
-			if unstructuredObj.GetKind() == string(addonsv1.SecretClusterResourceSetResourceKind) {
-				byteArr, _ = base64.StdEncoding.DecodeString(val)
-			}
 
-			dataList = append(dataList, byteArr)
+		if scopeError != nil {
+			errList = append(errList, scopeError)
+			continue
```

Comment:
Can we move this up after L301?
so we create the context, if err, continue, if !needApply, continue

Reply:
Yeah, we can. But that would behave differently as before, I only did this to keep the previous behavior.

Before this change, the controller couldn't fail before setting the binding since the only thing it did before was to check if the flag `applied` was true. Now, in order to check if the resource needs to be applied, we need (for `Reconcile` mode) to compute the hash, which could fail. So to keep the same behavior, we compute the hash, if it fails, we set the binding and `continue`.

Reply:
I think we should move this up because what is really important is that the controller is reentrant and as a consequence, the code should recover both from transient errors in creating the binding and in creating the scope.
In this case, I don't think the order of operation does not seem relevant for re-entrancy, so I think we should prefer better code readability/maintainability.

Reply:
Is this comment still relevant or is it related to this? https://github.com/kubernetes-sigs/cluster-api/pull/7497#discussion_r1082433155

Can we please re-create this comment? I can't find the location in the code we're talking about as code changed.

Reply:
@sbueringer I believe it's the same issue, resolving this one in favor of https://github.com/kubernetes-sigs/cluster-api/pull/7497#discussion_r1082433155